### PR TITLE
Make CoreBundle optional

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this changes are BC.

Part of https://github.com/sonata-project/dev-kit/issues/697

## Changelog

```markdown
### Removed
- remove SonataCoreBundle dependencies
```

### CODE REVIEW

File: src/Meta/MetadataInterface.php [hunk 1]
```diff
@@ -13,41 +13,80 @@
 
 namespace Sonata\BlockBundle\Meta;
 
-/**
- * @author Hugo Briand <briand@ekino.com>
- *
- * NEXT_MAJOR: Remove CoreBundle dependency
```
Comment (jordisala1991) [inline src/Meta/MetadataInterface.php:19]: This should be replaced by the BlockBundle MetadataInterface , or that is what the CoreBundle says

**Comment by jordisala1991** [comment]
@wbloszyk you should upgrade to sf4.3, remove di-extra-bundle

File: src/Meta/MetadataInterface.php [hunk 2]
```diff
@@ -13,41 +13,80 @@
...
- */
-interface MetadataInterface extends \Sonata\CoreBundle\Model\MetadataInterface
-{
+if (interface_exists('Sonata\CoreBundle\Model\MetadataInterface')) {
```
Comment (core23) [inline src/Meta/MetadataInterface.php:16]: Please use simple proxies

File: composer.json [hunk 3]
```diff
@@ -74,4 +77,4 @@
             "Sonata\\BlockBundle\\Tests\\": "tests/"
         }
     }
-}
+},
```
Comment (jordisala1991) [inline composer.json:80]: Remove comma

Comment (wbloszyk) [reply composer.json]: I saw it after request for review. Fixed.

File: src/SonataBlockBundle.php [hunk 4]
```diff
@@ -40,11 +40,13 @@ public function boot()
     /**
      * Register form mapping information.
      */
-    public function registerFormMapping()
+    private function registerFormMapping()
```
Comment (jordisala1991) [inline src/SonataBlockBundle.php:43]: Dont change the visibility , it is not needed

Comment (wbloszyk) [reply src/SonataBlockBundle.php]: Done.

**Comment by jordisala1991** [comment]
https://travis-ci.org/github/sonata-project/SonataBlockBundle/jobs/697315079

Travis does not pass

**Comment by wbloszyk** [comment]
@jordisala1991 You look at wrong request (this with `},` in the end)

https://travis-ci.org/github/sonata-project/SonataBlockBundle/builds/697317076

**Comment by jordisala1991** [comment]
Thank you @wbloszyk

**Pull request merged**